### PR TITLE
request barrier (testcafe part of the hammerhead #498 issue)

### PR DESCRIPTION
**Pull request by miherlosev**
@AlexanderMoskovkin @helen-dikareva

### CODE REVIEW

File: src/client/core/request-barrier.js [hunk 1]
```diff
@@ -30,46 +30,58 @@ export default class XhrBarrier extends EventEmitter {
...
+    _onRequestFinished (request) {
+        if (indexOf(this.requests, request) > -1) {
+            removeItem(this.requests, request);
 
             if (!this.collectingReqs && !this.requests.length)
                 this.emit(REQUESTS_FINISHED_EVENT);
         }
     }
 
+    _onFetchSend (request) {
```
Comment (AlexanderMoskovkin) [inline src/client/core/request-barrier.js]: As I understand we have the following common functions (for xhr and fetch): `_onRequestCompleted`, `_onRequestError` and `_onRequestFinished`. `Send` function are different for xhr and fetch, so I suppose it should be `_onXhrSend` and `_onFetchSend`

**Comment by AlexanderMoskovkin** [comment]
rebase to fix server tests.
and it seems, there are some problems with client-tests (https://travis-ci.org/testcafe-build-bot/testcafe/jobs/134437611, https://saucelabs.com/tests/af85838ca4594ed497b55c79cd04ad76)

File: src/client/core/request-barrier.js (hunk 1 above)
Comment (miherlosev) [inline src/client/core/request-barrier.js]: i don't understand what you mean
Need rename `_onRequestSend` to `_onXhrSend`?

Comment (AlexanderMoskovkin) [inline src/client/core/request-barrier.js]: yep

Comment (miherlosev) [inline src/client/core/request-barrier.js]: Done

**Comment by miherlosev** [comment]
> rebase to fix server tests.

Done

**Comment by miherlosev** [comment]
FPR

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by helen-dikareva** [comment]
lgtm

**Pull request merged**